allennlp/data/iterators/bucket_iterator.py

```diff
@@ -108,8 +217,6 @@ def __init__(
         maximum_samples_per_batch: Tuple[str, int] = None,
         skip_smaller_batches: bool = False,
     ) -> None:
-        if not sorting_keys:
```

Comment:
Hmm - i've actually come to a slightly different conclusion, which is that users assume that `BasicIterator` is a good starting point and do not discover that BucketIterator would be better. I'm generally against having arguments which duplicate checks like whether the sorting keys are none, because it's not super intuitive. Perhaps a warning is good here, if you don't pass any sorting keys?